db/migrate/20190906165501_remove_broken_variants_from_carts.rb

```diff
@@ -0,0 +1,15 @@
+class RemoveBrokenVariantsFromCarts < ActiveRecord::Migration
+  def up
+    # Removes line_items from open carts where the variant has a hard-deleted price
+
+    variants_without_prices = 'SELECT spree_variants.id FROM spree_variants
+      LEFT OUTER JOIN spree_prices ON (spree_variants.id = spree_prices.variant_id)
+      WHERE NOT EXISTS (SELECT 1 FROM spree_prices WHERE spree_prices.variant_id = spree_variants.id)'
+
+    broken_line_items = Spree::LineItem.
+      joins(:variant).where(variant: { id: variants_without_prices }).
```

Comment:
This migration is failing - It currently compares `variant.id` against the SQL subquery string above. I don't actually know how to use subqueries with the `where` method yet.

Reply:
Might be a Rails specific version :thinking: . But I tested in a Rails console and it correctly performed a single query using the subquery above.